graph/src/extension.rs

```diff
@@ -0,0 +1,120 @@
+use futures::sync::oneshot;
```

Comment:
The filename `extension.rs` could be an extension of anything. Can we rename it to e.g. `graph/src/ext/futures.rs` or `graph/src/util/futures.rs`?